apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/TextResponse/TextResponseFields/Label/Label.tsx

```diff
@@ -28,35 +33,61 @@ export const TEXT_RESPONSE_LABEL_UPDATE = gql`
 
 export function Label(): ReactElement {
   const { t } = useTranslation('apps-journeys-admin')
-  const [textResponseLabelUpdate] = useMutation<TextResponseLabelUpdate>(
-    TEXT_RESPONSE_LABEL_UPDATE
-  )
-  const { journey } = useJourney()
-  const { state } = useEditor()
+  const [textResponseLabelUpdate] = useMutation<
+    TextResponseLabelUpdate,
+    TextResponseLabelUpdateVariables
+  >(TEXT_RESPONSE_LABEL_UPDATE)
+  const { state, dispatch } = useEditor()
+  const { add } = useCommand()
 
   const selectedBlock = state.selectedBlock as
     | TreeBlock<TextResponseBlock>
     | undefined
 
+  function updateEditorState(
+    step: TreeBlock<BlockFields_StepBlock> | undefined,
+    block: TreeBlock
+  ): void {
+    dispatch({
+      type: 'SetSelectedStepAction',
+      selectedStep: step
+    })
+
+    dispatch({
+      type: 'SetSelectedBlockAction',
+      selectedBlock: block
+    })
+  }
+
   async function handleSubmit(e: FocusEvent): Promise<void> {
-    if (journey == null || selectedBlock == null) return
     const target = e.target as HTMLInputElement
-    await textResponseLabelUpdate({
-      variables: {
-        id: selectedBlock?.id,
-        journeyId: journey.id,
-        input: {
-          label: target.value
-        }
-      },
-      optimisticResponse: {
-        textResponseBlockUpdate: {
-          id: selectedBlock?.id,
-          __typename: 'TextResponseBlock',
-          label: target.value
+    const label = target.value
+    if (selectedBlock != null) {
```

Comment:
also move this back to line 42.
```suggestion
    if (selectedBlock == null)  return
```